activejob/lib/active_job/logging.rb

```diff
@@ -87,12 +87,25 @@ def queue_name(event)
         def args_info(job)
           if job.arguments.any?
             ' with arguments: ' +
-              job.arguments.map { |arg| arg.try(:to_global_id).try(:to_s) || arg.inspect }.join(', ')
+              job.arguments.map { |arg| serialize_arg(arg).inspect }.join(', ')
           else
             ''
           end
         end
 
+        def serialize_arg(arg)
```

Comment:
If we reuse `Arguments#serialize`, an exception will be raised once an unsupported argument type is encountered, therefore preventing the enqueuing from being properly logged. Are you sure you want it to work like that?